File: day1/validate_subsequence.py
# Time Complexity : O(n) - In worst case loops through the entire array of length n
# Space Complexity: O(1)- Storing only indexes
def isValidSubsequence(array, sequence):
    arrayIndex = 0
    seqIndex = 0
    while arrayIndex < len(array) and seqIndex < len(sequence):
        if(array[arrayIndex] == sequence[seqIndex]):
            seqIndex += 1
        arrayIndex += 1

    return len(sequence) == seqIndex

File: day1/validate_subsequence.java
/* Time Complexity : O(n) - In worst case loops through the entire array of length n
Space Complexity: O(1)- Storing only indexes */
import java.util.*;

class Program {
  public static boolean isValidSubsequence(List<Integer> array, List<Integer> sequence) {
    int arrayIndex = 0, seqIndex = 0;
    while(arrayIndex < array.size() && seqIndex < sequence.size()) {
      if(array.get(arrayIndex) == sequence.get(seqIndex)) {
        seqIndex++;
      }
      arrayIndex++;
    }
    if(seqIndex == sequence.size()) return true;
    return false;
  }
}